docusaurus/docs/e2e-test-a-plugin/getting-started.md

```diff
@@ -0,0 +1,497 @@
+---
+id: getting-started
+title: Getting started
+description: Getting started
+draft: true
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - testing
+  - e2e
+  - getting-started
+sidebar_position: 2
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Getting started
+
+This article will guide through how to install and configure `@grafana/plugin-e2e`, write tests and setup a basic Github workflow that can run your e2e tests targeting multiple versions of Grafana.
+
+## Prerequisites
+
+- You need to have a Grafana plugin [development environment](https://grafana.com/developers/plugin-tools/get-started/set-up-development-environment)
+- Node.js 18+
+- Docker
+- Basic Knowledge of Playwright. If you have not worked with Playwright before, we recommend following the [Getting started](https://playwright.dev/docs/intro) section in their documentation.
+
+### Installing Playwright
+
+Please refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install. `@grafana/plugin-e2e` extends Playwright APIs, so you need to have `Playwright/test` with a minimum version of 0.40.0 installed as a dev dependency in the package.json file of your plugin.
+
+## Set up `plugin-e2e`
+
+### Step 1: Installing @grafana/plugin-e2e
+
+Now open the terminal, cd into your plugin root folder and install `@grafana/plugin-e2e`.
+
+<Tabs
+defaultValue="npm">
+<TabItem value="npm">
+
+```bash
+npm install @grafana/plugin-e2e@latest --save-dev
+```
+
+</TabItem>
+
+<TabItem value="yarn">
+
+```bash
+yarn add @grafana/plugin-e2e@latest --dev
+```
+
+</TabItem>
+
+<TabItem value="pnpm">
+
+```bash
+pnpm add @grafana/plugin-e2e@latest -D
+```
+
+</TabItem>
+</Tabs>
+
+### Step 2: Configure Playwright
+
+Open the `playwright.config.[js|ts]` file that was generated when Playwright was installed.
+
+1. Change the `baseUrl` to `'http://localhost:3000'`.
+
+```ts title="playwright.config.ts"
+use: {
+  baseURL: 'http://localhost:3000',
+  ...
+}
+```
+
+2. Add a new `auth` setup project. This will login to Grafana and store the authenticated state on disk.
+
+```ts title="playwright.config.ts"
+projects: [
+  {
+    name: 'auth',
+    testDir: 'node_modules/@grafana/plugin-e2e/dist/auth',
+    testMatch: [/.*\.js/],
+  }
+```
+
+3. Then add another project that runs all the tests in a browser of choice. This project needs to depend on the `auth` project we added in the previous step. That will ensure login only happens once, and all tests in the `run-tests` project will start already authenticated.
+
+```ts title="playwright.config.ts"
+projects: [
+  {
+    name: 'auth',
+    testDir: 'node_modules/@grafana/plugin-e2e/dist/auth',
```

Comment:
This path might not be correct for PNPM due to its symlink strategy. It's a shame testDir doesn't resolve npm packages automagically. One solution here would be to use `require.resolve` to get the path...

```ts
const pluginE2eAuth = `${dirname(require.resolve('@grafana/plugin-e2e'))}/auth`;

// defineConfig bits...
 projects: [
    // 1. Login to Grafana and store the cookie on disk for use in other tests.
    {
      name: 'auth',
      testDir: pluginE2eAuth,
      testMatch: [/.*\.js/],
    },
```

Reply:
Works like a charm, thank you!